models/repo/pushmirror.go

```diff
@@ -63,34 +68,62 @@ func UpdatePushMirror(m *PushMirror) error {
 	return err
 }
 
-// DeletePushMirrorByID deletes a push-mirrors by ID
-func DeletePushMirrorByID(ID int64) error {
-	_, err := db.GetEngine(db.DefaultContext).ID(ID).Delete(&PushMirror{})
-	return err
-}
-
 // DeletePushMirrorsByRepoID deletes all push-mirrors by repoID
 func DeletePushMirrorsByRepoID(repoID int64) error {
 	_, err := db.GetEngine(db.DefaultContext).Delete(&PushMirror{RepoID: repoID})
 	return err
 }
 
-// GetPushMirrorByID returns push-mirror information.
-func GetPushMirrorByID(ID int64) (*PushMirror, error) {
-	m := &PushMirror{}
-	has, err := db.GetEngine(db.DefaultContext).ID(ID).Get(m)
+func DeletePushMirrors(opts PushMirrorOptions) error {
```

Comment:
You can create a new function to generate a `builder.Cond`, so that the function will become simple. You can find many examples like FindxxxxOptions in Gitea.